packages/dds/sequence/src/test/ddsFuzzHarness.ts

```diff
@@ -0,0 +1,505 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as path from "path";
+import { mkdirSync, readFileSync } from "fs";
+import { strict as assert } from "assert";
+import {
+	BaseFuzzTestState,
+	createFuzzDescribe,
+	defaultOptions,
+	done,
+	ExitBehavior,
+	Generator,
+	generatorFromArray,
+	interleave,
+	makeRandom,
+	performFuzzActions,
+	Reducer,
+	repeat,
+	SaveInfo,
+} from "@fluid-internal/stochastic-test-utils";
+import {
+	MockFluidDataStoreRuntime,
+	MockStorage,
+	MockContainerRuntimeFactoryForReconnection,
+	MockContainerRuntimeForReconnection,
+} from "@fluidframework/test-runtime-utils";
+import { IChannelFactory, IChannelServices } from "@fluidframework/datastore-definitions";
+import { unreachableCase } from "@fluidframework/common-utils";
+
+export interface Client<TChannelFactory extends IChannelFactory> {
+	channel: ReturnType<TChannelFactory["create"]>;
+	containerRuntime: MockContainerRuntimeForReconnection;
+}
+
+export interface DDSFuzzTestState<TChannelFactory extends IChannelFactory>
+	extends BaseFuzzTestState {
+	containerRuntimeFactory: MockContainerRuntimeFactoryForReconnection;
+	clients: Client<TChannelFactory>[];
+	// Client which was selected to perform an operation on
+	client: Client<TChannelFactory>;
+	// dds which was selected to perform an operation on. will be the same as client.channel
+	channel: ReturnType<TChannelFactory["create"]>;
+}
+
+export interface ClientSpec {
+	channelId: string;
+}
+
+export interface BaseOperation {
+	type: number | string;
+}
+
+export interface ChangeConnectionState {
+	type: "changeConnectionState";
+	connected: boolean;
+}
+
+export interface Synchronize {
+	type: "synchronize";
+}
+
+function getSaveDirectory(
+	model: DDSFuzzModel<any, any, any>,
+	options: DDSFuzzSuiteOptions,
+): string | undefined {
+	if (!options.saveFailures) {
+		return undefined;
+	}
+	const workloadFriendly = model.workloadName.replace(/[\s_]+/g, "-").toLowerCase();
+	return path.join(options.saveFailures.directory, workloadFriendly);
+}
+
+function getSaveInfo(
+	model: DDSFuzzModel<any, any, any>,
+	options: DDSFuzzSuiteOptions,
+	seed: number,
+): SaveInfo | undefined {
+	const directory = getSaveDirectory(model, options);
+	if (!directory) {
+		return undefined;
+	}
+	const filepath = path.join(directory, `${seed}.json`);
+	return { saveOnFailure: true, filepath };
+}
+
+/**
+ * @example
+ * Typical
```

Comment:
Did you mean to write more here or is this some shorthand I'm not familiar with?

Reply:
lol, yeah. definitely unfinished thought. thanks :)